ci/setup_gcs_cache.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+set -e
+
+gcp_service_account_cleanup() {
+  echo "Deleting service account key file..."
+  rm -rf /tmp/gcp_service_account.json
+}
+
+if [[ ! -z "${BAZEL_REMOTE_CACHE}" ]]; then
+
+  if [[ ! -z "${GCP_SERVICE_ACCOUNT_KEY}" ]]; then
+    echo "${GCP_SERVICE_ACCOUNT_KEY}" | base64 --decode > /tmp/gcp_service_account.json
+    trap gcp_service_account_cleanup EXIT
+
+    export BAZEL_BUILD_EXTRA_OPTIONS="${BAZEL_BUILD_EXTRA_OPTIONS} \
+      --remote_http_cache=${BAZEL_REMOTE_CACHE} --google_credentials=/tmp/gcp_service_account.json"
```

Comment:
Do we have to use filesystem? If so, best to make sure that the file perms are appropriately restricted.

Reply:
Yes, switched to using `mktemp` also addresses the permission.